lib/vty.c

```diff
@@ -1529,13 +1528,9 @@ static int vty_flush(struct thread *thread)
 	int vty_sock = THREAD_FD(thread);
 	struct vty *vty = THREAD_ARG(thread);
 
-	vty->t_write = NULL;
```

Comment:
This seems wrong to me:
If `VTYSH` is not defined then `t_write` will be set in `vty_event` on `event = VTY_WRITE` and will not be unset. Other places in code might check that variable and assume there is a `flush` going on when it might already happened.

I think you should remove `t_write = NULL` in `vty_write` and keep the one in `vty_flush` (since `vty_write` calls `vty_flush`) or even remove this different code path (just call `vty_flush` directly)!

Reply:
It's already set to NULL when we get here, that is why I removed it.